incubator/kafka/templates/jmx-configmap.yaml

```diff
@@ -10,7 +10,56 @@ metadata:
     chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
 data:
   jmx-kafka-prometheus.yml: |+
-{{- if .Values.metrics.jmx.kafkaConfig }}
-{{ toYaml .Values.metrics.jmx.kafkaConfig | indent 4 }}
+{{- if .Values.jmx.configMap.overrideConfig }}
+{{ toYaml .Values.jmx.configMap.overrideConfig | indent 4 }}
+{{- else }}
+    jmxUrl: service:jmx:rmi:///jndi/rmi://127.0.0.1:{{ .Values.jmx.port }}/jmxrmi
+    lowercaseOutputName: true
+    lowercaseOutputLabelNames: true
+    ssl: false
+    whitelistObjectNames: ["{{ join "\",\"" .Values.jmx.whitelistObjectNames }}"]
+    rules:
+    - pattern: kafka.controller<type=(ControllerChannelManager), name=(QueueSize), broker-id=(\d+)><>(Value)
+      name: kafka_controller_$1_$2_$4
+      labels:
+        broker_id: "$3"
+    - pattern: kafka.controller<type=(ControllerChannelManager), name=(TotalQueueSize)><>(Value)
+      name: kafka_controller_$1_$2_$3
+    - pattern: kafka.controller<type=(KafkaController), name=(.+)><>(Value)
+      name: kafka_controller_$1_$2_$3
+    - pattern: kafka.controller<type=(ControllerStats), name=(.+)><>(Count)
+      name: kafka_controller_$1_$2_$3
+    - pattern: kafka.server<type=(ReplicaFetcherManager), name=(.+), clientId=(.+)><>(Value)
+      name: kafka_server_$1_$2_$4
+      labels:
+        client_id: "$3"
+    - pattern : kafka.network<type=(Processor), name=(IdlePercent), networkProcessor=(.+)><>(Value)
+      name: kafka_network_$1_$2_$4
+      labels:
+        network_processor: $3
+    - pattern : kafka.network<type=(RequestMetrics), name=(RequestsPerSec), request=(.+)><>(Count)
+      name: kafka_network_$1_$2_$4
```

Comment:
On the previous PR I talked about why I'm not really a fan of rewriting metric names (https://github.com/kubernetes/charts/pull/4138#discussion_r180472559)

Granted, I am not really familiar with jmx configuration, but will using all of these rules as defaults potentially prevent users from being able to effectively use existing JMX/monitoring tutorials? (metrics not existing as they 'should', etc?)

I think providing education about how to configure this appropriately is good, but I'm not sure that complex rules should be part of the default. Perhaps this configuration information, or a link to it, would be best served as part of the NOTES.txt template?

Reply:
This is something I feel very strongly about.  The way metrics are exposed doesn't work well in Prometheus.  If you spend a few mins looking at the stats they are super ugly, doing things like moving topics to labels 💯 is something that should be done.

If you want to change some of the default jmx configuration I'm open, but it should ship with a standard that starts users off in a good place.

Reply:
Especially since I plan to publish a Grafana dashboard to https://grafana.com/dashboards and that's based on the formatting of the stats.  I by no means have all the answers to what should be monitored and best dashboards, but I want to provide a starting point and let the community help improve them (how to clean up stats too)

Reply:
I agree with your points, I'm just concerned that would then not be clear how to use the metrics without digging into the chart, or diving in to find the configuration applied. 

This seems like a good use of the `NOTES.txt` section; how about we add a note that default JMX configuration has been added, along with instructions of how to view it? (ie, a copy/paste-able `kubectl` command to view the configmap)

Reply:
I like that suggestion, anything we can do to be clear without making people dig.  

I'm actually trying to get jconsole (comes with java, gui I found with a quick google) connecting, b/c when I was creating the config file I was looking at raw metrics connection to `.Values.jmx.port` via jconsole and looking at scraped metrics in a web browser (port forwarding to access them).  So I'm trying to get that going again but having a little trouble, if I get it working I can try to put those details in NOTES.txt too.

Reply:
Nice suggestion on the `NOTES.txt` front, I think that will make things nice and clear for the end user